deployment/roles/monitor/tasks/cron.yml

```diff
@@ -16,3 +16,6 @@
     month: "{{ month }}"
     weekday: "{{ weekday }}"
     job: "{{ job }}"
+
+- name: Run the job to generate initial data
```

Comment:
They are not really needed. I tried to keep the behaviour that the monitor has data from the beginning and not wait until the cron job runs. I can remove it if you want

Reply:
1. Just worrying how much time could take deployment then if I am operating with monitor for three bridges (for example)
2. In order to be consistent in this case we need to re-generate the data after entire system/service restart. Otherwise the web serve could provide very old info.

Reply:
One possible option I thinking could be to remove running the job as a deployment task so it is not affected by the time consumed and add a call to generate the data in the `start` method of the service generated, so after the server is started it will generate the data also in a system/service restart.

For example update `start` in `deployment/roles/monitor/templates/tokenbridge-monitor.j2` to:
```bash
start(){
    echo "Starting TokenBridge Monitor.."
    cd $WORKDIR
    sudo -u "{{ compose_service_user }}" /usr/local/bin/docker-compose  down -v
    sudo -u "{{ compose_service_user }}" /usr/local/bin/docker-compose  rm -fv
    sudo -u "{{ compose_service_user }}" /usr/local/bin/docker-compose  up  --detach
    sudo -u "{{ compose_service_user }}" /bin/bash -c 'cd scripts; ./getBridgeStats.sh >cronWorker.out 2>cronWorker.err'
}
```
What do you think?

Reply:
Yes, it could behave in such manner.

Will the web service cause an error if someone will request data before the corresponding json file is generated?

Reply:
When you request data and there is not json file generated there is a default json returned defined as:
```json
{
      "error": "please check your worker"
}
```
Should we provide a more detailed error message?

Reply:
How about to re-phrase it to `the bridge statistics is not available`?

Reply:
Updated the logic to run the job on the service https://github.com/poanetwork/tokenbridge/pull/262/commits/d7fb9bc63e0374836e4cba13ef2887aaf1eca140.
And updated the error message https://github.com/poanetwork/tokenbridge/pull/262/commits/67132fd866fb9f941590bac675ace3938fc1fd21
I had to also add a call to run the checks as a post-deployment step in the monitor deployment e2e tests because the tests to check the files and endpoints start immediately after the deployment and the data was not ready https://github.com/poanetwork/tokenbridge/pull/262/commits/033201dc9cdb0ffadee9787ceaf02495b932323e